index.js

```diff
@@ -31,21 +31,32 @@ const server = http.createServer(async function (request, response) {
             gridPrecision = 5;
         }
 
+        const body = {
+            exact_bounds: true,
+            extent: 4096,
+            grid_precision: gridPrecision,
+            grid_type: 'grid',
+            size: params.renderMethod === 'hits' ? 10000 : 0,// only populate the hits layer when necessary
+            query: params.searchQuery ? { //use Lucene query_string syntax
+                "query_string": {
+                    "query": params.searchQuery,
+                    "analyze_wildcard": true
+                }
+            } : {
+                "match_all": {}
+            }
+        }
+
         try {
             const tile = await client.searchMvt({
                 index: params.index,
                 field: params.geometry,
                 zoom: parseInt(params.z),
                 x: parseInt(params.x),
                 y: parseInt(params.y),
-                exact_bounds: true,
-                extent: 4096,
                 grid_agg: params.renderMethod === 'grid' ? 'geotile' : 'geohex',
```

Comment:
Can you also move grid_agg and track_total_hits to body please

Reply:
Yep, done, sorry I missed those two from the docs.